docs/actionprovider.rst

```diff
@@ -1,25 +1,143 @@
 
-.. [1] The previous version of the Action Provider, deprecated April 2023,
-   had an ActionUrl of https://dev.funcx.org/automate along with
-   a required ``ActionScope``.  The current Action Provider is backwards
-   compatible, provided the ``ActionUrl`` is updated to the new URL and the
-   ``ActionScope`` removed.  (The ActionScope is now populated from the
-   Action Provider's metadata URL)
+Both Action Providers return their outputs in the same format. The V3 Action Provider
+generally supports more functionality than the V2 Action Provider, with some caveats;
+see below for details and migration.
 
+V3 Action Input Schema
+----------------------
 
-Action Input Schema
--------------------
+The V3 Action Provider supports all of the same input options as the |V3SubmitRoute|_
+in the Compute hosted web service, with two major differences. The V3 Action Provider
+accepts task input as JSON data, while the web service route only accepts task input as
+Compute-serialized strings. The V3 Action Provider also takes the endpoint ID as a JSON
+parameter, rather than in the URL.
 
-The Action Provider input schema accepts input for a single task, plus,
+A complete task submission might look like this:
+
+.. code-block:: json
+
+  {
+    "endpoint_id": "0c48c27a-0c2d-4028-a71b-d78697abbc4a",
+    "tasks": [
+      {
+        "function_id": "ff960aba-fa23-43d5-9cbe-3f4f91a066e1",
+        "args": [ "..." ],
+        "kwargs": { "...": "..." }
+      },
+      {
+        "function_id": "0a98fd06-edbd-11ed-abcd-0d705ebb4c49",
+        "args": [ "..." ],
+        "kwargs": { "...": "..." }
+      }
+    ],
+    "task_group_id": "97241626-8ff4-4550-9938-5909bd221869",
+    "user_endpoint_config": {
+      "min_blocks": 0,
+      "max_blocks": 1,
+      "scheduler_options": "#SBATCH --constraint=knl,quad,cache"
+    },
+    "resource_specification": {
+      "num_nodes": 2,
+      "ranks_per_node": 2,
+      "num_ranks": 4,
+      "launcher_options": "--cpu-bind quiet --mem 3072"
+    },
+    "create_queue": false
+  }
+
+The two required arguments are ``endpoint_id`` and ``tasks``. The former must be a valid
+UUID, and should point to an endpoint you have access to; the latter is a list of
+objects that represent the tasks to execute on that endpoint. Submissions must contain at
+least one task.
+
+Each task object must have a valid ``function_id``, while ``args`` and ``kwargs`` are
+optional (although if a function expects either, the flow will fail when the endpoint
+attempts to execute it). The ``args`` option is a list of arbitrary JSON data, and the
+``kwargs`` option is an object that maps argument keywords to argument values as
+arbitrary JSON data.
+
+For each task, the given endpoint will call the given function with any given arguments
+and keyword arguments, and the V3 Action Provider will return with the results of those
+calls.
+
+The other top-level arguments are optional - see the |V3SubmitRoute|_ for more
```

Comment:
To be a bit more clear, give a couple of examples? ie. 
``The other top-level arguments, such as task_group_id, user_endpoint_config, are optional - see the ...``